lib/extract-svg-metadata.js

```diff
@@ -0,0 +1,106 @@
+const createConfig = require('svgo/lib/svgo/config');
+const convertSVG = require('svgo/lib/svgo/svg2js');
+const applyPlugins = require('svgo/lib/svgo/plugins');
+const computePathBounds = require('svg-boundings').path;
+
+const extractMapboxIconMetadata = {
+    type: 'perItem',
+    fn: function(item, params, info) {
+        if (item.isElem('path') && item.hasAttr('id')) {
+            const id = item.attr('id');
+            if (id.value.startsWith('mapbox-icon') && item.hasAttr('d') && item.attr('d').value) {
+                const bounds = computePathBounds({ d: item.attr('d').value }, true);
+
+                // Make sure that we actually have a valid bounding box.
+                if (!Number.isFinite(bounds.left) || !Number.isFinite(bounds.right) ||
+                    !Number.isFinite(bounds.top) || !Number.isFinite(bounds.bottom)) {
+                    return true;
+                }
+
+                if (id.value.startsWith('mapbox-icon-stretch-x')) {
+                    (info['stretch-x'] = info['stretch-x'] || []).push([bounds.left, bounds.right]);
+                } else if (id.value.startsWith('mapbox-icon-stretch-y')) {
+                    (info['stretch-y'] = info['stretch-y'] || []).push([bounds.top, bounds.bottom]);
+                } else if (id.value === 'mapbox-icon-stretch') {
+                    (info['stretch-x'] = info['stretch-x'] || []).push([bounds.left, bounds.right]);
+                    (info['stretch-y'] = info['stretch-y'] || []).push([bounds.top, bounds.bottom]);
+                } else if (id.value === 'mapbox-icon-content') {
+                    info.content = [ bounds.left, bounds.top, bounds.right, bounds.bottom ];
+                }
+            }
+        }
+        return true;
+    }
+};
+
+const config = createConfig({
+    full: true,
+    plugins: [
+        'convertShapeToPath',
+        'convertTransform',
+        'moveGroupAttrsToElems',
+        'collapseGroups',
+        'convertPathData',
+        { extractMapboxIconMetadata }
+    ]
+});
+
+function sortStretchByFirstCoordinate(a, b) {
+    return a[0] - b[0];
+}
+
+/**
+ * Parses a SVG document and extracts metadata from its shapes and paths.
+ *
+ * @param   {string}                [svg]                 A string of the SVG.
+ * @param   {Function}              callback              Accepts two arguments, `err` and `metadata` Object
+ * @return  {Metadata}              metadata              An object with the extracted information.
+ */
+function extractMetadata(svg, callback) {
+    try {
+        convertSVG(svg, function(result) {
+            if (result.error) {
+                return callback(result.error);
+            }
+
+            const info = {};
```

Comment:
A comment about how this const gets mutated while running applyPlugins would be helpful.